File: src/main/org/nlogo/prim/etc/_pendown.java
package org.nlogo.prim.etc ;

import org.nlogo.agent.Turtle;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Syntax;

public final strictfp class _pendown
	extends Command
{
	@Override
	public Syntax syntax()
	{
		return Syntax.commandSyntax( "-T--" , true ) ;
	}
	@Override
	public void perform( final org.nlogo.nvm.Context context )
	{
		( (Turtle) context.agent ).penMode( Turtle.PEN_DOWN ) ;
		context.ip = next ;
	}
}

File: src/main/org/nlogo/prim/etc/_pendown.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.agent.Turtle
import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Command, Context }

class _pendown extends Command {
  override def syntax =
    Syntax.commandSyntax("-T--", true)
  override def perform(context: Context) {
    context.agent.asInstanceOf[Turtle].penMode(Turtle.PEN_DOWN)
    context.ip = next
  }
}
